backend/modules/student/routers/student_instances.py

```
@@ -908,3 +908,125 @@ async def start_simulation_from_instance(
+            logger.info(f"Linked new progress to instance {instance_unique_id}: user_progress_id={result.user_progress_id}")
+
+        return {
+            "success": True,
+            "message": "Simulation reset successfully",
+            "user_progress_id": result.user_progress_id,
+            "simulation": result.simulation,
+            "current_scene": result.current_scene,
+            "simulation_status": result.simulation_status,
+            "all_scenes": result.all_scenes,
+            "attempts_count": instance.attempts_count if instance else 1
+        }
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)